tartiflette/schema/schema.py

```diff
@@ -629,10 +643,14 @@ def _validate_directive_implementation(self) -> List[str]:
         for directive in self._directive_definitions.values():
             for expected in _IMPLEMENTABLE_DIRECTIVE_HOOKS:
                 attr = getattr(directive.implementation, expected, None)
-                if attr and not is_valid_coroutine(attr):
+                if (
+                    attr
+                    and not is_valid_coroutine(attr)
+                    and not is_valid_async_generator(attr)
```

Comment:
Nones of directive hooks specified in `_IMPLEMENTABLE_DIRECTIVE_HOOKS` should be an `AsyncGenerator`. I'm not sure to understand this new check.

Reply:
on_schema_subscription is an AsyncGenerator.

Reply:
My bad, when I checked, I didn't see `on_schema_subscription` into the `_IMPLEMENTABLE_DIRECTIVE_HOOKS` constant 👍

In the future maybe we could add two checks, one for async generator and one for coroutine with two distincts whitelist constants (in order to make sure that we do have async generator and coroutine where they are expected).

Reply:
Yep, good point i'll do it now, no point in waiting